src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarWriter.Unix.cs

```diff
@@ -58,8 +58,8 @@ public sealed partial class TarWriter : IDisposable
                 entry._header._devMinor = (int)minor;
             }
 
-            entry._header._mTime = TarHelpers.GetDateTimeOffsetFromSecondsSinceEpoch(status.MTime);
-            entry._header._aTime = TarHelpers.GetDateTimeOffsetFromSecondsSinceEpoch(status.ATime);
+            entry._header._mTime = info.LastWriteTimeUtc;
+            entry._header._aTime = info.LastAccessTimeUtc;
             entry._header._cTime = TarHelpers.GetDateTimeOffsetFromSecondsSinceEpoch(status.CTime);
```

Comment:
From looking at the FileStatus code:

[https://github.com/dotnet/runtime/blob/fdc3b51da075066a5da489be2c169be85dbc117f/src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.Unix.cs#L309-L318](https://github.com/dotnet/runtime/blob/fdc3b51da075066a5da489be2c169be85dbc117f/src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.Unix.cs#L309-L318)

It looks like `status.XTime` gets added to `status.XTimeNsec` to make this DateTimeOffset. We will be respecting that for M and A time, but it looks like we don't respect that for C time now.

Reply:
Good point. Changing this logic to:

```cs
            entry._header._cTime = DateTimeOffset.FromUnixTimeSeconds(status.CTime).AddTicks(status.CTimeNsec / 100 /* nanoseconds per tick */);
```